packages/@webex/internal-plugin-scheduler/src/scheduler/scheduler.js

```diff
@@ -49,108 +43,247 @@ class Scheduler extends WebexPlugin {
+      );
 
-    // Complex example
-    return this.request({
-      uri: 'https://www.example.com', // a URI can be used in place of service/resource.
-      method: 'POST',
-      headers: {
-        accept: 'application/json',
-        authorization: 'example-token',
-        'x-custom-header': 'x-custom-header-example-value',
-      },
-      body: {
-        'example-param-a': 'example-param-a-value',
-        'example-param-b': 'example-param-b-value',
-        'example-param-c': 'example-param-c-value',
-      },
-    });
-  }
+      return Promise.reject(new Error('SDK cannot authorize'));
+    }
+
+    if (this.registered) {
+      this.logger.info('internal-plugin-scheduler -> register#INFO, plugin already registered');
+
+      return Promise.resolve();
+    }
+
+    return this.webex.internal.device
+      .register()
+      .then(() => this.webex.internal.mercury.connect())
+      .then(() => {
+        this.listenForEvents();
+        this.trigger(CONSTANTS.SCHEDULER_REGISTERED);
+        this.registered = true;
+      })
+      .catch((error) => {
+        this.logger.error(
+          `internal-plugin-scheduler -> register#ERROR, Unable to register, ${error.message}`
+        );
+
+        return Promise.reject(error);
+      });
+  },
 
   /**
-   * Example event usage. Note that an event engine is mapped to `this`
-   * upon extending the `WebexPlugin` class constructor. This includes
-   * the following methods:
-   * `this.listenTo()`, `this.stopListening()`, `this.trigger()`, `this.on()`, etc.
-   *
-   * Note that all methods provided as event handlers should be associated with
-   * a namespace so that they can be referenced/destroyed if/when necessary.
+   * Explicitly tears down the scheduler plugin by unregister
+   * the device, disconnecting from mercury, and stops listening to calendar events
    *
-   * @returns {void}
+   * @returns {Promise}
+   * @public
+   * @memberof Calendar
    */
-  exampleEventUsage() {
-    // listen for locally scoped events [using on, internal]
-    this.on('event:scope', (event) => {
-      this.logger.log(event);
-    });
+  unregister() {
+    if (!this.registered) {
+      this.logger.info(
+        'internal-plugin-scheduler -> unregister#INFO, Calendar plugin already unregistered'
+      );
 
-    // stop listening for locally scoped events.
-    this.off('event:scope', () => {
-      /* use previous `on` method param namespace instead of arrow function */
+      return Promise.resolve();
+    }
+
+    this.stopListeningForEvents();
+
+    return this.webex.internal.mercury
+      .disconnect()
+      .then(() => this.webex.internal.device.unregister())
+      .then(() => {
+        this.trigger(CONSTANTS.SCHEDULER_UNREGISTERED);
+        this.registered = false;
+      });
+  },
+
+  /**
+   * registers for calendar events through mercury
+   * @returns {undefined}
+   * @private
+   */
+  listenForEvents() {
```

Comment:
Assuming `listenForEvents()` will be updated to handle more events at a later time?

i.e.

```js
// repeated for each event to emit
this.webex.internal.mercury.on('example:event', (envelope) => {
  this.trigger('{calendar/scheduler}:event-name', envelope);
});
  ```